### ISSUE TITLE
[RFC] New dependency: react-native-swiper

### ISSUE THREAD

**Issue by dariakoko** [artsy/eigen#7158]
#### Status 
Adopted

#### New Dependency
Name: react-native-swiper
URL: https://github.com/leecade/react-native-swiper

### Focus

A React Native module that allows you to quickly build customisable Swiper components.

Check out the implementation and test out PR [here](https://github.com/artsy/eigen/pull/7156). 

#### Check List
- [x] Have read over the source code, and we can maintain it
- [x] Has had a release in the last year or looks done and stable

#### Alternatives
`react-native-swiper` seems to be the easiest solution for the purpose of the task (swipe between slides that display data and, based on the index of the card, update the slide-specific data). The Swiper component makes it easier to operate with indexes of the slides and child components. 
Before testing `react-native-swiper`, I've tried out using SrollView. The implementation seemed to be not as transparent and robust as `react-native-swiper`, plus the ScrollView renders all child components at once, which adds more complexity to future tasks. With `react-native-swiper` we can render only the current slide.

**Comment by lordkiz** [comment]
I don't see why you cannot use this library. 
> Before testing react-native-swiper, I've tried out using SrollView. The implementation seemed to be not as transparent and robust as react-native-swiper, plus the ScrollView renders all child components at once, which adds more complexity to future tasks. With react-native-swiper we can render only the current slide.

But usually ViewPagers are used for data that isn't large. So Scrollview in this case would not have really bad effect on performance, if that is your major concern. Infact this library now uses Scrollview to implement its Pager. The longer route would be to use FlatList with `pagingEnabled` and detect current index of Page in view using `onViewableItemsChanged`. But that logic is already implemented in this lib.

In summary I vote that you use this lib and avoid reimplementing.

**Comment by MounirDhahri** [comment]
I love `react-native-swiper` I suggest we go for it as well. Thanks for opening this RFC

**Comment by araujobarret** [comment]
My only concern regarding this lib is it had no maintenance over the past 2 years 😢

**Comment by damassi** [comment]
+1 to @araujobarret! That's a long time. And it's always worth looking at [the number of issues](https://github.com/leecade/react-native-swiper/issues) and [open pull requests](https://github.com/leecade/react-native-swiper/pulls). 667 issues and over a 100 pull requests; that's a liability if something doesn't quite work. 

@dariakoko and @MounirDhahri, is there something that is better maintained that achieves the same functionality?

**Comment by pvinis** [comment]
thats a good point.

what do we want to use this for?
we have used just flatlist for this in folio.

**Comment by dariakoko** [comment]
@MounirDhahri what do you think?
The lib works and works well for our scenario
Implementing the Scrollview again will take some time, I had some performance issues when trying it last time

I would merge the solution with the library to unblock the release. In the future, when might consider writing our own component

**Comment by damassi** [comment]
> I would merge the solution with the library to unblock the release

@dariakoko - but is there another solution that is maintained? We shouldn't rush when it involves dependencies that are unmaintained.

**Comment by MounirDhahri** [comment]
Actually react-native-swiper is more like stable than unmaintained, it's become kind of the defacto go-to solution for swipers in react-native. The Flatlist/Scrollview API hasn't changed much over the past few years and that explains why not much has been added to react native swiper.

**Comment by dariakoko** [comment]
Closing this RFS.
We agreed not to use this library because of all of the mentioned reasons and went with a ScrollView instead 👍
